dbms/programs/server/TCPHandler.cpp

```diff
@@ -185,6 +185,43 @@ void TCPHandler::runImpl()
                 state.maybe_compressed_in.reset(); /// For more accurate accounting by MemoryTracker.
             });
 
+            /// Send structure of columns to client for function input()
+            query_context->setInputInitializer([this] (Context & context, const StoragePtr & input_storage)
+            {
+                if (&context != &*query_context)
```

Comment:
&* looks weird.
If it's a shared or unique ptr or optional, you can use `.get()`.